File: Python/121-bestTimeToSellStock.py
class Solution:
    # @param {integer[]} prices
    # @return {integer}
    def maxProfit(self, prices):
        max_diff = 0
        min_price = float('inf')
        for i in xrange(len(prices)):
            if prices[i] < min_price:
               min_price = prices[i]
            if prices[i] - min_price > max_diff:
               max_diff = prices[i] - min_price
        return max_diff

if __name__ == '__main__':
   s = Solution()
   print s.maxProfit([1, 2, 4])

File: C++/121-BestTimeToSellStock.cpp
#include "include/common.h"

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.empty()) {
            return 0;
        };
        int min = prices[0], maxProfit = 0;
        for(int i = 0; i < prices.size(); i++) {
            if (min > prices[i]) {
                min = prices[i];
            };
            if (maxProfit < prices[i] - min) {
                maxProfit = prices[i] - min;
            }
        };
        return maxProfit;
    }
};

int main(void) {
    Solution s;
    int arr [] = {4, 5, 2, 6, 9};
    vector<int> iarr(arr, arr + 5);
    cout << s.maxProfit(iarr) << endl;
    int arr1 [] = {9, 2};
    vector<int> iarr1(arr1, arr1 + 2);
    cout << s.maxProfit(iarr1) << endl;
    int arr2 [] = {9};
    vector<int> iarr2(arr2, arr2 + 1);
    cout << s.maxProfit(iarr2) << endl;
    return 0;
};